### ISSUE TITLE
stress: failed test in cockroach/sql/sql.test: TestRaceWithBackfill

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#6148]
Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/250d0afc7fa8dc7fb8a0a1800e10035c3e2e54a1

Stress build found a failed test:

```
=== RUN   TestRaceWithBackfill
W160419 06:39:59.951040 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
I160419 06:39:59.951645 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
W160419 06:39:59.952816 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
I160419 06:39:59.953007 server/node.go:360  store store=0:0 ([]=) not bootstrapped
I160419 06:39:59.954619 storage/replica_command.go:1465  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m0.954308797s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:39:59.954523158 +0000 UTC]
I160419 06:39:59.955196 server/node.go:310  **** cluster {6403d915-14a4-413c-a87b-8d48ac600065} has been created
I160419 06:39:59.955221 server/node.go:311  **** add additional nodes by specifying --join=127.0.0.1:40707
W160419 06:39:59.955231 gossip/gossip.go:888  not connected to cluster; use --join to specify a connected node
I160419 06:39:59.955980 server/node.go:373  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:8109277184 RangeCount:0}
I160419 06:39:59.956012 server/node.go:285  node ID 1 initialized
I160419 06:39:59.956072 storage/stores.go:287  read 0 node addresses from persistent storage
I160419 06:39:59.956133 server/node.go:494  connecting to gossip network to verify cluster ID...
I160419 06:39:59.956711 server/node.go:515  node connected via gossip and verified as part of cluster {"6403d915-14a4-413c-a87b-8d48ac600065"}
I160419 06:39:59.956748 server/node.go:338  [node=1] Started node with [[]=] engine(s) and attributes []
I160419 06:39:59.956776 server/server.go:366  starting https server at 127.0.0.1:36996
I160419 06:39:59.956797 server/server.go:367  starting grpc/postgres server at 127.0.0.1:40707
I160419 06:39:59.956837 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160419 06:39:59.957641 storage/replica_command.go:1919  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160419 06:39:59.958772 server/updates.go:147  No previous updates check time.
I160419 06:39:59.965979 storage/replica_command.go:1465  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.055501031s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:39:59.965933302 +0000 UTC]
I160419 06:39:59.966099 storage/replica_command.go:1919  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160419 06:39:59.972322 storage/replica_command.go:1465  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.055501031s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:39:59.972282718 +0000 UTC]
I160419 06:39:59.972516 storage/replica_command.go:1919  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
I160419 06:39:59.980197 storage/replica_command.go:1465  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.055501031s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:39:59.980144582 +0000 UTC]
I160419 06:39:59.980497 storage/replica_command.go:1919  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160419 06:40:00.065265 storage/replica_command.go:1465  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.113538287s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:40:00.065198533 +0000 UTC]
I160419 06:40:00.068080 storage/split_queue.go:100  splitting range=5 [/Table/14-/Max) at keys [/Table/50/0 /Table/51/0]
I160419 06:40:00.068462 storage/replica_command.go:1919  initiating a split of range=5 [/Table/14-/Max) at key /Table/50
I160419 06:40:00.080083 storage/replica_command.go:1465  range 6: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.129762788s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:40:00.08003072 +0000 UTC]
I160419 06:40:00.090893 storage/replica_command.go:1919  initiating a split of range=6 [/Table/50-/Max) at key /Table/51
I160419 06:40:00.430216 storage/replica_command.go:1465  range 7: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 405846h40m1.479909225s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-04-19 06:40:00.430168669 +0000 UTC]
I160419 06:40:00.650639 sql/lease.go:211  publish (count leases): descID=51 name=test version=1 count=1
I160419 06:40:00.677895 sql/lease.go:211  publish (count leases): descID=51 name=test version=2 count=1
I160419 06:40:00.953077 gossip/gossip.go:914  starting client to 127.0.0.1:40707
I160419 06:40:00.953611 gossip/client.go:83  closing client to node 1 (127.0.0.1:40707): gossip/client.go:177: stopping outgoing client to node 1 (127.0.0.1:40707); loopback connection
I160419 06:40:01.226386 sql/lease.go:211  publish (count leases): descID=51 name=test version=3 count=1
I160419 06:40:01.256597 sql/lease.go:211  publish (count leases): descID=51 name=test version=4 count=1
I160419 06:40:01.278603 sql/lease.go:211  publish (count leases): descID=51 name=test version=5 count=1
I160419 06:40:01.622837 sql/lease.go:211  publish (count leases): descID=51 name=test version=6 count=1
W160419 06:40:01.624676 sql/lease.go:645  /Table/3/1/51/2/1: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:742: node 1 stopped
W160419 06:40:01.644251 sql/schema_changer.go:163  failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:742: node 1 stopped
W160419 06:40:01.644380 sql/schema_changer.go:149  failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:742: node 1 stopped
W160419 06:40:01.644515 sql/session.go:301  failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:742: node 1 stopped
W160419 06:40:01.644622 sql/session.go:301  failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:742: node 1 stopped
E160419 06:40:01.644661 sql/pgwire/v3.go:139  write tcp 127.0.0.1:40707->127.0.0.1:40844: use of closed network connection
I160419 06:40:02.624894 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:40707: getsockopt: connection refused"; Reconnecting to "127.0.0.1:40707"
I160419 06:40:02.644597 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:40707: getsockopt: connection refused"; Reconnecting to "127.0.0.1:40707"
I160419 06:40:03.963879 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:40707: getsockopt: connection refused"; Reconnecting to "127.0.0.1:40707"
I160419 06:40:03.964048 /go/src/google.golang.org/grpc/clientconn.go:317  Failed to dial 127.0.0.1:40707: grpc: timed out trying to connect; please retry.
I160419 06:40:04.274171 /go/src/google.golang.org/grpc/clientconn.go:462  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:40707: getsockopt: connection refused"; Reconnecting to "127.0.0.1:40707"
I160419 06:40:04.274446 /go/src/google.golang.org/grpc/clientconn.go:317  Failed to dial 127.0.0.1:40707: grpc: timed out trying to connect; please retry.
--- FAIL: TestRaceWithBackfill (4.38s)
	schema_changer_test.go:527: pq: index-id "2" does not exist
	schema_changer_test.go:467: sql: database is closed
	schema_changer_test.go:469: schema changes took 983.689591ms
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
1 runs so far, 0 failures, over 35s
4 runs completed, 1 failures, over 40s
FAIL
```
Please assign, take a look and update the issue accordingly.

**Comment by dt** [comment]
This was probably fixed by #6334 (though it still fails under stress if the box gets slow enough for a `context deadline exceeded`).

**Comment by vivekmenezes** [comment]
Where are you seeing it? on your own machine?

**Comment by vivekmenezes** [comment]
also see  https://github.com/cockroachdb/cockroach/issues/6333

**Comment by dt** [comment]
Yeah, my GCE instance with: `make stress PKG=./sql TESTS=TestRaceWithBackfill STRESSFLAGS="-p 64"`

**Comment by bdarnell** [comment]
This isn't quite the same message as #6432, but between the PR that fixed that bug and #6334, has this been fixed?

**Comment by vivekmenezes** [comment]
same as  #6432
